dohproxy/utils.py

```diff
@@ -41,6 +42,32 @@ def extract_path_params(url: str) -> Tuple[str, Dict[str, List[str]]]:
     return p.path, params
 
 
+def create_custom_ssl_context(
+        insecure: bool,
+        cafile: Optional[str] = None) -> ssl.SSLContext:
+    """ Create a custom SSL context
+    :param insecure: Disable certificate verification if True
+    :param cafile: Pass custom CA file for cert verification
+    :return: An instance of ssl.SSLContext with our configuration
+    """
+
+    if insecure:
+        sslctx = ssl.SSLContext(ssl.PROTOCOL_SSLv23)
+        sslctx.options |= ssl.OP_NO_SSLv2
+        sslctx.options |= ssl.OP_NO_SSLv3
+        sslctx.options |= ssl.OP_NO_COMPRESSION
+        sslctx.set_default_verify_paths()
+    else:
+        sslctx = ssl.create_default_context()
```

Comment:
`create_default_context` suports the cafile, capath... parameters. Can't we have those defined in `create_custom_ssl_context` signature and use them directly in `create_default_context` and bypass `load_verify_locations`.

see https://docs.python.org/3/library/ssl.html#ssl.create_default_context 

make all default to None and we are good.
Also, in python3 we could take advantage of the bare `*` to make sure the parameters in `create_custom_ssl_context`  are used as keyword arguments.

Reply:
The reason I didn't do it that way is that it creates an SSLContext with only the specified CA cert from the command line however, if we create the default context and then use *load_verify_locations* then the CAs in the context will be all the system CAs as well as the one the user defined on the command line which I thought is what we want.

Reply:
Hm, I might be wrong in that but it's definitely making a difference: 

```python
In [26]: sslctx = ssl.create_default_context()

In [27]: sslctx.load_verify_locations(cafile="/etc/ssl/cert.pem")

In [28]: sslctx.cert_store_stats()
Out[28]: {'crl': 0, 'x509': 177, 'x509_ca': 177}

In [29]: ssl.create_default_context(cafile="/etc/ssl/cert.pem").cert_store_stats()
Out[29]: {'crl': 0, 'x509': 48, 'x509_ca': 48}
```

Reply:
oh I see. Fair enough. so maybe last nit is to not use `capath` (unless we were actually using it.